src/components/Line/index.jsx

```diff
@@ -0,0 +1,87 @@
+import { PureComponent } from 'react';
+import { arrayOf, bool, func, number, object, shape, string } from 'prop-types';
+import LineNumber from '../LineNumber';
+import LineContent from '../LineContent';
+import {
+  line,
+  lineHighlight,
+  lineHover,
+  lineSelectable,
+} from './index.module.css';
+
+/**
+ * A single row of content, containing both the line number and any text content within the line.
+ */
+export default class Line extends PureComponent {
+  static propTypes = {
+    data: arrayOf(
+      shape({
+        text: string,
+      })
+    ).isRequired,
+    number: number.isRequired,
+    rowHeight: number.isRequired,
+    highlight: bool,
+    selectable: bool,
+    style: object,
+    formatPart: func,
+    onLineNumberClick: func,
+    onRowClick: func,
+  };
+
+  static defaultProps = {
+    highlight: false,
+    selectable: false,
+    style: {},
+    formatPart: null,
+    onLineNumberClick: null,
+    onRowClick: null,
+  };
+
+  handleHover = ({ currentTarget, type }) => {
+    currentTarget.classList.toggle(lineHover, type === 'mouseenter');
+  };
+
+  render() {
+    const {
+      data,
+      formatPart,
+      highlight,
+      selectable,
+      onLineNumberClick,
+      onRowClick,
+      number,
+      rowHeight,
+      style,
+    } = this.props;
+    const highlightClass = highlight ? ` ${lineHighlight}` : '';
+    const selectableClass = selectable ? ` ${lineSelectable}` : '';
+    const className = `${line}${highlightClass}${selectableClass}`;
+    const lineStyle = {
+      ...style,
+      lineHeight: `${style ? style.height || rowHeight : rowHeight}px`,
+      minWidth: style ? style.width || '100%' : '100%',
```

Comment:
Did you mean:

```js
minWidth: style ? style.width : '100%',
```

Reply:
No, It's still possible that `width` isn't defined in `style` so we still want to default to `100%`.